docs/topics/forms/formsets.txt

```diff
@@ -821,7 +891,21 @@ deal with the management form:
         </table>
     </form>
 
-The above ends up calling the ``as_table`` method on the formset class.
+The above ends up calling the :meth:`BaseFormSet.render()` method on the
+formset class. This renders the formset using the template specified by the
+:attr:`~BaseFormSet.template_name` attribute. Similar to forms, by default the
+formset will be rendered ``as_table``, with other helper methods of ``as_p``
+and ``as_ul`` being available. The rendering of the formset can be customized
+by specifying the ``template_name`` attribute, or more generally by
+:ref:`overriding the default template <overriding-built-in-formset-templates>`.
+
+.. versionchanged:: 4.0
+
+    Formsets are now rendered using the template engine. To enable this
+    :attr:`BaseFormSet.renderer`, :attr:`BaseFormSet.template_name`, 
+    :attr:`BaseFormSet.template_name_p`, :attr:`BaseFormSet.template_name_ul`,
+    :attr:`BaseFormSet.template_name_table`, :meth:`BaseFormSet.render()`, and
+    :meth:`BaseFormSet.get_context()` were added.
```

Comment:
I'd add `versionadded` annotations to all new attributes and methods and change it to the:
```suggestion
     Rendering of formsets was moved to the template engine.
```